tests/utils.py

```diff
@@ -0,0 +1,128 @@
+import itertools
+import random
+import string
+
+import avro.schema
+
+
+def valueFromSetGenerator(possibilities=[]):
+    while True:
+        yield random.choice(possibilities)
```

Comment:
My only (slight) concern here is the problem of reproducibility. When we use a global random generator, the values we use for a given test will depend on the initial random seed, and the order and number of tests that we run. If we use an `InstanceGenerator` which has a private instance of a random generator that we seed, then we can run these tests in any order or combination and be guaranteed to get the same results. 

Not sure if this is worth worrying about, but I thought I'd bring it up in any case.